### PR TITLE
Using reporter middleware options for a11y test results

### PR DESCRIPTION
**Pull request by MelSumner**
### :pushpin: Summary

If merged, this PR updates the a11y test invocation strategy

### Details

- updates the YAML file for the action
- adds the middleware report options to the test-helpers
- adds a `test-a11y-report` script to the package.json
- adds `ember-a11y-report` folder to the npmignore file
- uploads the results of running the test with the middleware turned on, as json file

### CODE REVIEW

File: .github/workflows/ci-components-nightly.yml [hunk 1]
```diff
@@ -35,17 +35,17 @@ jobs:
...
+        run: yarn run test:a11y-report
         working-directory: packages/components
         continue-on-error: true
       
       - name: Archive a11y test results
         uses: actions/upload-artifact@v3
         with:
           name: a11y-test-results
-          path: packages/components/test-output/a11y-test-results-${{ env.DATE }}.txt
+          path: packages/components/ember-a11y-testing/figure-out-date-format-in-name.json
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply .github/workflows/ci-components-nightly.yml:49]: I suggest that we don't try to match the file name, it's going to be just one so even the line below will do (essentially using a wildcard)
```suggestion
          path: packages/components/ember-a11y-testing/*.json
```

**Comment by MelSumner** [comment]
ok @alex-ju I think this might be ready to try again if the wildcard json approach works. I think there's still probably some iterating to do because I think we want the results committed to the folder in the repo (?) but this will get us one step closer.

**Review by alex-ju — APPROVED** [review]
If we're not going to have another action to pick up the artifact (likely so, considering the meeting we had today), then yes, we'll want to commit these reports to the repo (or write them somewhere). Cristiano is doing something very similar for the metrics project, so hopefully we'll be able to use that code, so I wouldn't worry about it right now. In the meantime, was it intentional that you committed the report? I'm happy to have it there if useful, just wanted to make sure it was a conscious decision.

**Comment by MelSumner** [comment]
Yes, intentional for the time being; made sure to add the folder to npmignore so it isn't included in any builds. 

Ok going to try running again and see where we land. 👍

**Pull request merged**